aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -687,6 +691,197 @@ public void onError(Exception error) {
         );
     }
 
+    @Override
+    public void fetchUserAttributes(
+            @NonNull Consumer<List<AuthUserAttribute>> onSuccess,
+            @NonNull Consumer<AuthException> onError
+    ) {
+        awsMobileClient.getUserAttributes(new Callback<Map<String, String>>() {
+            @Override
+            public void onResult(Map<String, String> result) {
+                List<AuthUserAttribute> userAttributes = new ArrayList<>();
+                for (String userAttributeKey : result.keySet()) {
+                    userAttributes.add(new AuthUserAttribute(
+                            new AuthUserAttributeKey(userAttributeKey),
+                            result.get(userAttributeKey)
+                    ));
+                }
+                onSuccess.accept(userAttributes);
+            }
+
+            @Override
+            public void onError(Exception error) {
+                onError.accept(new AuthException(
+                        "Failed to fetch user attributes",
+                        error,
+                        "See attached exception for details"
+                ));
+            }
+        });
+    }
+
+    @Override
+    public void updateUserAttribute(
+            @NonNull AuthUserAttribute attribute,
+            @NonNull Consumer<AuthUpdateAttributeResult> onSuccess,
+            @NonNull Consumer<AuthException> onError
+    ) {
+
+        Map<String, String> userAttribute = new HashMap<>();
+        userAttribute.put(
+                attribute.getKey().getKeyString(),
+                attribute.getValue()
+        );
+
+        awsMobileClient.updateUserAttributes(
+                userAttribute,
+                new Callback<List<UserCodeDeliveryDetails>>() {
+                @Override
+                public void onResult(List<UserCodeDeliveryDetails> result) {
+                    if (result != null) {
+                        for (UserCodeDeliveryDetails details : result) {
+                            if (details.getAttributeName().equals(attribute.getKey().getKeyString())) {
+                                onSuccess.accept(new AuthUpdateAttributeResult(
+                                        true,
+                                        new AuthNextUpdateAttributeStep(
+                                                AuthUpdateAttributeStep.CONFIRM_ATTRIBUTE_WITH_CODE,
```

Comment:
This is correct if there's a UserCodeDeliveryDetails object returned only for attributes that are being confirmed. If it is returned in an empty state or something like that for attributes that don't need confirmation this is wrong. Most attributes will simply be done after you call update attribute without needing a confirm state.